tests/test_functons.py

```diff
@@ -12,56 +12,88 @@
 
 
 @pytest.mark.parametrize("test_square_input,test_square_expected", [
+    (-(3037000499**2-1), -sys.maxsize-1),
+    (-(3037000499**2), -sys.maxsize),
     (-10, 100),
     (-1, 1),
     (0, 0),
     (1, 1),
-    (2, 4),
     (10, 100),
+    (3037000499**2-1, sys.maxsize-1),
+    (3037000499**2, sys.maxsize)
 ])
 def test_square(test_square_input, test_square_expected):
     assert square(test_square_input) == test_square_expected
 
 
 @pytest.mark.parametrize("test_input,expected", [
-    (-100, -99),
+    (-sys.maxsize,  -sys.maxsize+1),
+    (-sys.maxsize-1,  -sys.maxsize),
+    (-3037000499, -3037000498),
     (-1, 0),
     (0, 1),
     (1, 2),
-    (100, 101),
+    (3037000499, 3037000500),
+    (sys.maxsize-1, sys.maxsize),
+    (sys.maxsize, sys.maxsize+1)
 ])
 def test_one_good_turn(test_input, expected):
     assert one_good_turn(test_input) == expected
 
 
 @pytest.mark.parametrize("test_square_expected,test_deserves_another_expected", [
-    (-100, -97),
+    (-sys.maxsize,  -sys.maxsize+3),
+    (-sys.maxsize-1,  -sys.maxsize+2),
+    (-3037000499, -3037000496),
```

Comment:
also what about `-sys.maxsize**2` - to test a kind of a middle value of the partition `-infinity` to `-sys.maxsize-1`?